site/README.md

```diff
@@ -0,0 +1,182 @@
+
+The non-versioned site pages are all the `/site/docs/.*md` files and the docs are the `/site/docs/docs/<version>/docs/*.md` files. Notice the location of the `mkdocs.yml`. Looking at this though, you may ask where the older versions and javadocs are.
+
+```
+./site/
+├── docs
+│   ├── assets
+│   ├── docs
+│   │   └── nightly
+│   │       ├── docs
+│   │       │   ├── assets
+│   │       │   ├── api.md
+│   │       │   ├── ...
+│   │       │   └── table-migration.md
+│   │       └── mkdocs.yml (versioned)
+│   ├── about.md
+│   ├── ...
+│   └── view-spec.md
+├── README.md
+├── mkdocs.yml (non-versioned)
+├── requirements.txt
+└── variables.yml
+```
+### Building the versioned docs
+
+All previously versioned docs will be committed in `docs-<version>` branches and mounted using [git worktree](https://git-scm.com/docs/git-worktree) at build time. The worktree will pull these versions in following the `/site/docs/docs/<version>` convention. The `latest` version, will be a secondary copy of the most recent build version in the worktree, but pointing to `/site/docs/docs/latest`. There is also a `javadoc` branch that contains all prior static generation versions of the javadocs in a single tag.
+
+```
+./site/
+└── docs
+    ├── docs
+    │   ├── nightly
+    │   ├── latest
+    │   ├── 1.3.1
+    │   ├── 1.3.0
+    │   └── ...
+    └── javadoc
+        ├── latest
+        ├── 1.3.1
+        ├── 1.3.0
+        └── ...
+```
+
+### Install
+
+1. (Optional) Set up venv
+```
+python -m venv mkdocs_env
+source mkdocs_env/bin/activate
+```
+
+1. Install required Python libraries
+```
+pip install -r requirements.txt
+```
+
+#### Adding additional versioned documentation
+
+To build locally with additional docs versions, add them to your working tree.
+For now, I'm just adding a single version, and the javadocs directory.
+
+```
+git worktree add site/docs/docs/1.3.1 docs-1.3.1
+git worktree add site/docs/javadoc javadoc
+```
+
+## Build
+
+Run the build command in the root directory, and optionally add `--clean` to force MkDocs to clear previously generated pages.
+
+```
+mkdocs build [--clean]
+```
+
+## Run
+
+Start MkDocs server locally to verify the site looks good.
+
+```
+mkdocs serve
+```
+
+## Release process
+
+Deploying a version of the docs is a two step process:
+ 1. Cut a new release from the current branch revision. This creates a new branch `docs-<version>`,.
+
+    ```
+    .github/bin/deploy_docs.sh -v 1.4.0
```

Comment:
We should move this to maybe a `dev` folder with scripts rather than `.github` which seems like an odd place for it.

Reply:
Added a note here as well to ignore while we remove the script.